overlord/snapstate/snapmgr.go

```diff
@@ -488,10 +480,17 @@ func (m *SnapManager) NextRefresh() time.Time {
 	return m.nextRefresh
 }
 
-// RefreshSchedule returns the current refresh schedule.
+// RefreshSchedule returns the current refresh schedule as a string
+// suitable to display to a user.
 // The caller should be holding the state lock.
 func (m *SnapManager) RefreshSchedule() string {
-	return m.currentRefreshSchedule
+	_, scheduleStr, err := m.refreshSchedule()
+	// refreshSchedule should almost never return an error, except
+	// when the state is corrupted or in similar emergencies.
+	if err != nil {
+		return fmt.Sprintf("cannot get refresh.schedule: %v", err)
```

Comment:
Why isn't this a separate error return value?

Reply:
Thanks for pointing this out, the reasons are stupid, I did not want to change the call site but thats silly of course, thanks again for nudging it to be a clean solution rather than a nasty mess :)